### PR TITLE
feat: face hog v01

### PR DESCRIPTION
**Pull request by pauldambra**
For the Rome 2022 hackathon

## Problem

You don't know when other people change your flags or insights

## Changes

Well, now you do!

* Adds an endpoint to get changes to your insights or feature flags from the activity log _if you didn't make them_
* and a notification bell so you can see them
* polls for notifications for that dopamine rush
 
### unread notification

Screenshot 2022-10-01 at 08 25 58

## How did you test this code?

running it locally

### CODE REVIEW

File: frontend/src/layout/navigation/TopBar/notificationsLogic.ts [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+    loaders(({ actions, values }) => ({
+        importantChanges: [
+            [] as HumanizedActivityLogItem[],
+            {
+                loadImportantChanges: async (_, breakpoint) => {
+                    await breakpoint(1)
+
+                    clearTimeout(values.pollTimeout)
+
+                    const response = (await api.get(
```
Comment (benjackwhite) [inline frontend/src/layout/navigation/TopBar/notificationsLogic.ts:30]: If the network request fails can this throw? If so do we rather want to wrap it in a try/finally and have the timeout recreation in the finally block?

File: frontend/src/lib/constants.tsx [hunk 2]
```diff
@@ -124,6 +124,7 @@ export const FEATURE_FLAGS = {
     FEATURE_FLAGS_UX: 'feature-flags-ux', //owner: @liyiy
     REGION_SELECT: 'region-select', //owner: @kappa90
     INGESTION_WARNINGS_ENABLED: 'ingestion-warnings-enabled', // owner: @macobo
+    HOG_BOOK: 'hog-book', // owner: @pauldambra
```
Comment (benjackwhite) [inline frontend/src/lib/constants.tsx:127]: lol

**Review by benjackwhite — APPROVED** [review]
Looks great! Awesome that we can sort of just drop in the historic activity components here. I didn't go super thorough on my own testing but as it is behind a FF I think it'll be best to test in the wild

File: frontend/src/layout/navigation/TopBar/NotificationsBell.scss [hunk 3]
```diff
@@ -0,0 +1,4 @@
+.NotificationsBell-Popup {
+    z-index: 949; // so TZ aware time pop-up is not behind this pop-up
```
Comment (Twixes) [inline frontend/src/layout/navigation/TopBar/NotificationsBell.scss:2]: AFAIK we're keeping track of z-indexes in `vars.scss`

**Review by Twixes — COMMENTED** [review]
Code-wise this looks pretty good though I've gotta say there are major UX/product questions I think we'll need to answer with notifications like this before rollout. 😅 I'm most worried that if all changes in the project (org org) result in a notification, notifications will lose value for users because of an extremely low signal to noise ratio. Both notifications and an audit log seem valuable, but they should probably be each handled slightly differently.

File: frontend/src/layout/navigation/TopBar/NotificationsBell.scss (hunk 3 above)
Comment (pauldambra) [reply frontend/src/layout/navigation/TopBar/NotificationsBell.scss:2]: https://github.com/PostHog/posthog/pull/12087

**Review by mariusandra — COMMENTED** [review]
Found a few things to remark, otherwise looks good enough to try out.

UX: bug with large text.


UX: this image makes me want to click it. Let's not use the attention-grabbing "blue active" color to say "nothing to see here"
image

The concerns I do have are around clarity of what is shown. As a user, how am I supposed to know what kind of events show up in this notification bar, and what to expect for. How can I configure it? As a developer, how do I add more stuff? 

Implementation wise, I'm also not sure if polling every 5 sec is the lightest thing to do, but it's for sure good enough for a flagged demo.

**Comment by pauldambra** [comment]
various follow-up PRs

* https://github.com/PostHog/posthog/pull/12093
* https://github.com/PostHog/posthog/pull/12092
* https://github.com/PostHog/posthog/pull/12090
* https://github.com/PostHog/posthog/pull/12089
* https://github.com/PostHog/posthog/pull/12088
* https://github.com/PostHog/posthog/pull/12087
* https://github.com/PostHog/posthog/pull/12086

**Pull request merged**